### PR TITLE
p11sak: User friendly error when mechanism or key size is not supported

### PR DESCRIPTION
**Pull request by ifranzki**
Before attempting to generate a key using a mechanism, check if that mechanism is supported by the token in the selected slot, and if the key size is supported. Display a user friendly error message if not.

Closes: https://github.com/opencryptoki/opencryptoki/issues/485

### CODE REVIEW

**Review by juergenchrist — APPROVED** [review]
lgtm

File: usr/sbin/p11sak/p11sak.c [hunk 1]
```diff
@@ -651,69 +652,89 @@ static CK_RV read_rsa_args(CK_ULONG modulusbits, CK_ULONG exponent,
...
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP320r1);
+        *keybits = 320;
     } else if (strcmp(ECcurve, "brainpoolP320t1") == 0) {
         pubattr[*pubcount].pValue = (CK_BYTE*) brainpoolP320t1;
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP320t1);
+        *keybits = 320;
     } else if (strcmp(ECcurve, "brainpoolP384r1") == 0) {
         pubattr[*pubcount].pValue = (CK_BYTE*) brainpoolP384r1;
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP384r1);
+        *keybits = 284;
```
Comment (jschmidb) [inline usr/sbin/p11sak/p11sak.c:725]: 384

File: usr/sbin/p11sak/p11sak.c [hunk 2]
```diff
@@ -651,69 +652,89 @@ static CK_RV read_rsa_args(CK_ULONG modulusbits, CK_ULONG exponent,
...
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP384t1);
+        *keybits = 384;
     } else if (strcmp(ECcurve, "brainpoolP512r1") == 0) {
         pubattr[*pubcount].pValue = (CK_BYTE*) brainpoolP512r1;
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP512r1);
+        *keybits = 521;
     } else if (strcmp(ECcurve, "brainpoolP512t1") == 0) {
         pubattr[*pubcount].pValue = (CK_BYTE*) brainpoolP512t1;
         pubattr[*pubcount].ulValueLen = sizeof(brainpoolP512t1);
+        *keybits = 521;
```
Comment (jschmidb) [inline usr/sbin/p11sak/p11sak.c:737]: 512?

**Review by jschmidb — APPROVED** [review]
lgtm

File: usr/sbin/p11sak/p11sak.c (hunk 1 above)
Comment (ifranzki) [reply usr/sbin/p11sak/p11sak.c:725]: wow, thank you, you have good eys :-)

File: usr/sbin/p11sak/p11sak.c (hunk 2 above)
Comment (ifranzki) [reply usr/sbin/p11sak/p11sak.c:737]: yep

**Review by finn-callies — APPROVED** [review]

**Pull request merged**